lib/web_ui/lib/src/engine/platform_views/content_manager.dart

```diff
@@ -212,9 +212,10 @@ class PlatformViewManager {
   /// component.
   bool isVisible(int viewId) => !isInvisible(viewId);
 
-  /// Clears the state. Used in tests.
-  ///
   /// Returns the set of know view ids, so they can be cleaned up.
+  Set<int> getKnownPlatformViewIds() => _contents.keys.toSet();
```

Comment:
Does this need to be a `Set`? What if we just use `Iterable<int>` and get rid of the extra step of converting to a `Set`?
```suggestion
  Iterable<int> getKnownPlatformViewIds() => _contents.keys;
```

Reply:
This is a `Set` because that's what the `disposeViews` method of the CK renderer wants:

* https://github.com/flutter/engine/blob/4020f0ee4dafe8df963280912720fb9a1b2bc004/lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart#L513

(That method wants a Set because most operations in that file are Set intersections/differences IIRC)

Reply:
This got me thinking.. since `HtmlViewEmbedder` already keeps track of platform views, why not use that instead of calling `PlatformViewManager.instance.getKnownPlatformViewIds()`. And then it dawned on me that `HtmlViewEmbedder` is per-view but `PlatformViewManager` is a singleton.

This means: `FlutterView.dispose()` => `ViewRasterizer.dispose()` => `HtmlViewEmbedder.dispose()` => `dispose all views that PlatformViewManager knows about`.

That said, it's working fine today because even though `HtmlViewEmbedder` iterates over _**all**_ platform views, it doesn't do anything dangerous to the platform views that are unrelated to itself. But why iterate over _**all**_ platform views (inefficient)? And it's super easy to accidentally change `disposeViews()` and it starts doing something dangerous to those unrelated platform views (e.g. remove some html attribute).

I suggest we find a way to iterate over only the platform views that are related to this instance of `HtmlViewEmbedder`.

Reply:
Yes, I definitely double-checked that the embedder only acted on views it knew about, but we could definitely keep track of those so we never have to touch the view manager (or only to dispose views there, from the embedder)

Reply:
This is totally doable, I'm just using the keys of the maps from which it'll **do anything** anyways. That lets us isolate this class.

Reply:
I tweaked this quite a bit!